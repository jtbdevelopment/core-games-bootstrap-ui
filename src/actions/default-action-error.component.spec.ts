import {TestBed} from '@angular/core/testing';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {DefaultActionErrorComponent} from './default-action-error.component';


class MockModal {
    dismiss = jasmine.createSpy('dismiss');
}

describe('Component:  default action error component', () => {
    let fixture;
    let modal;
    beforeEach(() => {
        TestBed.configureTestingModule({
            declarations: [
                DefaultActionErrorComponent,
            ],
            providers: [
                {provide: NgbActiveModal, useClass: MockModal}
            ],
        });
        TestBed.compileComponents();
        fixture = TestBed.createComponent(DefaultActionErrorComponent);
        modal = TestBed.get(NgbActiveModal);
        fixture.detectChanges();
    });

    it('displays error message', () => {
        fixture.componentInstance.errorMessage = 'Something went wrong!';
        fixture.detectChanges();
        expect(fixture.nativeElement.querySelector('.error-message').textContent.trim()).toEqual('Something went wrong!');
    });

    it('cancel modal', () => {
        expect(modal.dismiss).not.toHaveBeenCalled();
        fixture.nativeElement.querySelector('.close-button').click();
        expect(modal.dismiss).toHaveBeenCalled();
    });
});
