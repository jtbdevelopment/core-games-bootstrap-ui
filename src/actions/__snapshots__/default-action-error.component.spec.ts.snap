// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  default action error component displays error message 1`] = `
<ngbd-modal-content
  activeModal={[Function MockModal]}
  errorMessage={[Function String]}
>
  <div
    class="game-error-dialog"
    role="dialog"
  >
    
    
    <div
      class="modal-header"
    >
      
      
      <h4
        class="modal-title"
      >
        Sorry!
      </h4>
      
    
    </div>
    
    
    <div
      class="modal-body"
    >
      
      
      <span
        class="error-message"
      >
        Something went wrong!
      </span>
      
    
    </div>
    
    
    <div
      class="modal-footer"
    >
      
      
      <button
        class="btn btn-info focus close-button"
      >
        OK
      
      </button>
      
    
    </div>
    
  
  </div>
  
</ngbd-modal-content>
`;
